src/utils/Utils.ts

```diff
@@ -434,4 +434,8 @@ export class Utils {
     addDiff(secondObject, firstObject);
     return difference;
   }
+
+  static waitUntil<T = void>(ms: number, returns?: T): Promise<T> {
+    return new Promise(resolve => setTimeout(() => (returns ? resolve(returns) : resolve()), ms));
```

Comment:
I would imagine that `waitUntil` would be used like this:

```
const value = 'value'
await waitUntil(1000)
// use value in some way
```

If so, is the `returns` argument needed considering you could access the value after waiting?

If it is needed, you could remove the ternary and just pass the `returns` arg no matter what. e.g.

```
return new Promise(resolve => setTimeout(() => resolve(returns), ms));
```

Reply:
I only added it because there's two main cases that come up a few times where I need a timed promise.
1. In a component, I want to delay something (like your example).
2. In a unit test, I want to promisify a result (e.g., `suggestionsManager.mergeSuggestions([Utils.waitUntil(10, someSuggestions)])`).

What do you think would be the best course of action here?
Should I still change it like you said? If so, should I make a new util for the second case scenario or not?

Reply:
Maybe "waitUntil" is just not the correct wording ? 

It's kind of a "race" function instead ?

Reply:
@btaillon, in your example, why do we need to add a 10ms wait if we already have the suggestions?
If the goal is to promisify, could we do:
```
suggestionsManager.mergeSuggestions([Promise.resolve(someSuggestions)])
```

Reply:
@olamothe I'm a bit confused by what you mean by `race`, as I see it, it's a delayed `Promise.resolve`. Would something like `delay()`, `resolveAfter()`, or `resolveIn` be a better name?

@samisayegh Unfortunately the goal isn't just to promisify, it's also to be able to test what happens in-between different promises.